setuptools/_core_metadata.py

```diff
@@ -62,7 +62,7 @@ def _read_list_from_msg(msg: Message, field: str) -> Optional[List[str]]:
 
 
 def _read_payload_from_msg(msg: Message) -> Optional[str]:
-    value = msg.get_payload().strip()
+    value = str(msg.get_payload()).strip()
```

Comment:
idk if this needs to be publicised as it all seems pretty internal. but this now supports `Message` whose payload is also a single-part `Message`. (multi-part Message support is undefined here, and will use a list repr)

Reply:
I believe that `is_multipart()` should always be false in this case, so according to the docs the output is a string... So in terms of runtime the function should be safe.

Adding a explicit call to `str` should be OK to convince the type checker.